test/lit/passes/signature-pruning.wast

```diff
@@ -1151,3 +1151,52 @@
     )
   )
 )
+
+;; $exported is exported. The entire rec group becomes exported as well, which
+;; causes $unused-param's type to be public, which means we cannot normally
+;; modify it. However, in closed world we allow such changes, and we can remove
+;; the unused param there. What happens is that we keep the original public rec
+;; group as-is, and add a new rec group for private types, put the pruned type
+;; there, and use that pruned type on $unused-param.
+(module
+  (rec
+   ;; CHECK:      (rec
+   ;; CHECK-NEXT:  (type $much (func))
+
+   ;; CHECK:       (type $1 (func))
+
+   ;; CHECK:      (rec
+   ;; CHECK-NEXT:  (type $none (func))
+   (type $none (func))
+   ;; CHECK:       (type $much (func (param i32)))
```

Comment:
It looks like the type name is repeated in the output, which we shouldn't allow.

Reply:
Yeah, that's a quirk we have atm. It's actually useful in cases where the old and new types remain in use (like here), but we should probably ensure a new unique name. I can look into that separately.

Reply:
I would expect us to have problems with text round trip fuzzing before this is fixed.

Reply:
I don't see issues in fuzzing so far, but this is a pre-existing issue, so that isn't surprising. It must take specific fuzzer luck to get it.

Reply:
Experimenting with a patch that deduplicates the names, the problem is that it will generate a large diff on existing tests (before this PR; unrelated) regardless of whether we deduplicate the old or the new name. We seem to just have enough cases that both remain in use. The diff one way is 66 K and the other is 276 K, so at least one is clearly less annoying, but it will still be a quite large change unfortunately.